lib/ProductOpener/APIProductServices.pm

```diff
@@ -107,69 +107,99 @@ my %service_functions = (
 	extend_ingredients => \&ProductOpener::Ingredients::extend_ingredients_service,
 	estimate_ingredients_percent => \&ProductOpener::Ingredients::estimate_ingredients_percent_service,
 	analyze_ingredients => \&ProductOpener::Ingredients::analyze_ingredients_service,
+	check_quality => \&check_quality_service,
 );
 
 sub check_product_services_api_input ($request_ref) {
 
-	my $response_ref = $request_ref->{api_response};
-	my $request_body_ref = $request_ref->{body_json};
-
-	my $error = 0;
-
-	# Check that we have an input body
-	if (not defined $request_body_ref) {
-		$log->error("product_services_api - missing or invalid input body", {}) if $log->is_error();
-		$error = 1;
-	}
-	else {
-		# Check that we have the input body fields we expect
-
-		if (not defined $request_body_ref->{product}) {
-			$log->error("product_services_api - missing input product", {request_body => $request_body_ref})
-				if $log->is_error();
-			add_error(
-				$response_ref,
-				{
-					message => {id => "missing_field"},
-					field => {id => "product"},
-					impact => {id => "failure"},
-				}
-			);
-			$error = 1;
-		}
-
-		if (not defined $request_body_ref->{services}) {
-			$log->error("product_services_api - missing services", {request_body => $request_body_ref})
-				if $log->is_error();
-			add_error(
-				$response_ref,
-				{
-					message => {id => "missing_field"},
-					field => {id => "services"},
-					impact => {id => "failure"},
-				}
-			);
-			$error = 1;
-		}
-		elsif (ref($request_body_ref->{services}) ne 'ARRAY') {
-			add_error(
-				$response_ref,
-				{
-					message => {id => "invalid_type_must_be_array"},
-					field => {id => "services"},
-					impact => {id => "failure"},
-				}
-			);
-			$error = 1;
-		}
-		else {
-			# Echo back the services that were requested
-			$response_ref->{services} = $request_body_ref->{services};
-		}
-	}
-	return $error;
+    my $response_ref = $request_ref->{api_response};
```

Comment:
For Perl files, we use tabs, they shouldn't be converted to spaces. (depending on your editor, it might pick it from the .editorconfig file)